test/new_relic/agent/log_event_aggregator_test.rb

```diff
@@ -0,0 +1,270 @@
+      end
+    end
+
+    def test_record_without_transaction_randomizes
+      # There can be only one
+      with_config(CAPACITY_KEY => 1) do
+        LogPriority.stubs(:rand).returns(0.9)
+        @aggregator.record("Buggy", "DEBUG")
+
+        LogPriority.stubs(:rand).returns(0.1)
+        @aggregator.record("Deadly", "FATAL")
+
+        metadata, results = @aggregator.harvest!
+
+        assert_equal(2, metadata[:events_seen])
+        assert_equal(1, metadata[:reservoir_size])
+        assert_equal(1, results.size)
+        assert_equal("Buggy", results.first.last["message"])
+      end
+    end
+
+    def test_lowering_limit_truncates_buffer
+      orig_max_samples = NewRelic::Agent.config[CAPACITY_KEY]
+
+      orig_max_samples.times do |i|
+        @aggregator.record("Truncation happens", "WARN")
+      end
+
+      new_max_samples = orig_max_samples - 10
+      with_config(CAPACITY_KEY => new_max_samples) do
+        metadata, results = @aggregator.harvest!
+        assert_equal(new_max_samples, metadata[:reservoir_size])
+        assert_equal(orig_max_samples, metadata[:events_seen])
+        assert_equal(new_max_samples, results.size)
+      end
+    end
+
+    def test_record_adds_timestamp
+      t0 = Process.clock_gettime(Process::CLOCK_REALTIME) * 1000
+      message = "Time keeps slippin' away"
+      @aggregator.record(message, "INFO")
+
+      _, events = @aggregator.harvest!
+
+      assert_equal(1, events.size)
+      event = events.first
+
+      assert_equal({
+          'level' => "INFO",
+          'message' => message,
+          'timestamp' => t0,
+        },
+        event.last)
+    end
+
+    def test_records_metrics_on_harvest
+      with_config CAPACITY_KEY => 5 do
+        9.times { @aggregator.record("Are you counting this?", "DEBUG") }
+        @aggregator.harvest!
+
+        assert_metrics_recorded_exclusive({
+          "Logging/lines" => { :call_count => 9 },
+          "Logging/lines/DEBUG" => { :call_count => 9 },
+          "Logging/Forwarding/Dropped" => { :call_count => 4 },
+          "Supportability/Logging/Forwarding/Seen" => { :call_count => 9 },
+          "Supportability/Logging/Forwarding/Sent" => { :call_count => 5 },
+        },
+        :ignore_filter => %r{^Supportability/API/})
+      end
+    end
+
+    def test_high_security_mode
+      with_config CAPACITY_KEY => 5, :high_security => true do
+
+        # We refresh the high security setting on this notification
+        NewRelic::Agent.config.notify_server_source_added
+
+        9.times { @aggregator.record("Are you counting this?", "DEBUG") }
+        _, items = @aggregator.harvest!
+
+        # Never aggregate logs
+        assert_empty items
+
+        # We are fine to count them, though....
+        assert_metrics_recorded_exclusive({
+          "Logging/lines" => { :call_count => 9 },
+          "Logging/lines/DEBUG" => { :call_count => 9 },
+          "Supportability/Logging/Metrics/Ruby/disabled" => { :call_count => 1 },
```

Comment:
The Engineering Brief currently states the default value for `application_logging.metrics.enabled` is `true`. This would change the metric on line 236 to end with `enabled` instead of `disabled`. Is this accurate for high security mode as well?

Reply:
This test isn't quite right because of the unfinished config work we should look at.

Metrics and local decoration configs just don't exist, so that's why they're disabled and 💯  agreed that these metrics should be `enabled`, even under high security when those exist.

The `enabled` on Forwarding below is reading the literal config value vs interpreting it with regards to high security mode, which I _think_ is the intent but maybe deserves double-checking.